backend/service/k8s/clientset.go

```diff
@@ -95,6 +107,25 @@ func newClientsetManager(rules *clientcmd.ClientConfigLoadingRules, logger *zap.
 	return &managerImpl{clientsets: lookup}, nil
 }
 
+func applyRestClientConfig(restConfig *restclient.Config, restClientConfig k8sconfigv1.RestClientConfig) error {
```

Comment:
take ptr to RestClientConfig and add nil check

also probably want to export this func so we can use in private